static/src/javascripts/projects/common/modules/ui/selection-sharing.js

```diff
@@ -0,0 +1,173 @@
+        <ul class="social u-unstyled u-cf" data-component="social-selection">
+            <li class="social__item" data-link-name="twitter">
+                <a  class="rounded-icon social-icon social-icon--twitter js-selection-twitter"
+                    data-link-name="social-selection"
+                    target="_blank"
+                    href="#">
+                    <span class="u-h">Share on Twitter</span>
+                    <%= twitterIcon %>
+                </a>
+            </li>
+            <li class="social__item" data-link-name="email">
+                <a  class="rounded-icon social-icon social-icon--email js-selection-email"
+                    data-link-name="social-selection"
+                    target="_blank"
+                    href="#">
+                    <span class="u-h">Share via Email</span>
+                    <%= emailIcon %>
+                </a>
+            </li>
+        </ul>
+    </div>
+`;
+
+const twitterIcon = inlineSvg('shareTwitter', ['icon', 'centered-icon']);
+const emailIcon = inlineSvg('shareEmail', ['icon', 'centered-icon']);
+const $body = document.body;
+
+const selectionSharing = template(sharingTemplate, {
+    twitterIcon,
+    emailIcon,
+});
+
+const $selectionSharing = $.create(selectionSharing);
+
+let $twitterAction;
+let $emailAction;
+const twitterShortUrl = `${config.page.shortUrl}/stw`;
+const twitterHrefTemplate =
+    'https://twitter.com/intent/tweet?text=%E2%80%9C<%=text%>%E2%80%9D&url=<%=url%>';
+
+const // 140 - t.co length - 3 chars for quotes and url spacing
+twitterMessageLimit = 114;
+
+const emailShortUrl = `${config.page.shortUrl}/sbl`;
+const emailHrefTemplate =
+    'mailto:?subject=<%=subject%>&body=%E2%80%9C<%=selection%>%E2%80%9D <%=url%>';
+const validAncestors = [
+    'js-article__body',
+    'content__standfirst',
+    'block',
+    'caption--main',
+    'content__headline',
+];
+
+const isValidSelection = (
+    range
+): boolean => // commonAncestorContainer is buggy, can't use it here.
+    validAncestors.some(
+        className =>
+            $.ancestor(range.startContainer, className) &&
+            $.ancestor(range.endContainer, className)
+    );
+
+const hideSelection = (): void => {
+    if ($selectionSharing.hasClass('selection-sharing--active')) {
+        $selectionSharing.removeClass('selection-sharing--active');
+    }
+};
+
+const showSelection = (): void => {
+    if (!$selectionSharing.hasClass('selection-sharing--active')) {
+        $selectionSharing.addClass('selection-sharing--active');
+    }
+};
+
+const updateSelection = (): void => {
+    const selection =
+        window.getSelection && document.createRange && window.getSelection();
+    let range;
+    let rect;
+    let top;
+    let twitterMessage;
+    let twitterHref;
+    let emailHref;
+
+    if (
+        $body &&
+        selection &&
+        selection.rangeCount > 0 &&
+        selection.toString()
+    ) {
+        range = selection.getRangeAt(0);
```

Comment:
Could all the assignments 

https://github.com/guardian/frontend/blob/980137bae5fb8c157787beca33842d047a784cd3/static/src/javascripts/projects/common/modules/ui/selection-sharing.js#L92-L97

be done here as constants?